### ISSUE TITLE
Incorrect JSON string representation when querying or exporting tables

### ISSUE THREAD

**Issue by trenta3** [duckdb/duckdb#7123]
### What happens?

JSON values are exported or returned in queries in an incorrect simil-JSON format which makes it difficult to use when using clients, or impossible to export tables to json format.

Briefly the returned format encloses keys in single quotes (instead of double quotes) and completely removes quotes from string values made of alphanumeric characters.

I expect a JSON field to return properly formatted JSON when querying.

### To Reproduce

```sql
INSTALL json;
LOAD json;
CREATE TABLE test (a JSON);
INSERT INTO test (a) VALUES ({'a': 'bc', 'd': 3.0 });
COPY test TO 'entries.json' (DELIMITER ',');
```
outputs `Error: Invalid Input Error: Malformed JSON at byte 1 of input: unexpected character.  Input: {'a': bc, 'd': 3.0}`.

Analogously, when querying from the Python client, the returned data is difficult to parse, as the format is a non-standard one.

### OS:

Linux x64 (Fedora 36)

### DuckDB Version:

v0.7.2-dev1034 18142dcc4b

### DuckDB Client:

CLI

### Full Name:

Dario Balboni

### Affiliation:

Scuola Normale Superiore (University)

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by trenta3** [comment]
I've now seen issue #4303 that is about a way to correct the problem at insertion time.
I've tried in converting the data already in the database using `SELECT to_json(a) FROM test` but I've received the error `Error: Invalid Input Error: Malformed JSON at byte 1 of input: unexpected character.  Input: {'a': bc, 'd': 3.0}`, which is a bit strange to me, as it seems like the database tries to parse the values as JSON raising the error, but this doesn't happen in normal querying.

**Comment by lnkuiper** [comment]
Duplicate of #6687 

What is happening here is that your table is of type `JSON`, and you are inserting a `STRUCT`. This auto-casts it to a string, which might not be valid `JSON`. Instead, try calling `to_json` before casting to `JSON`, i.e.:

```sql
D CREATE TABLE test (a JSON);
D INSERT INTO test (a) VALUES (to_json({'a': 'bc', 'd': 3.0 }));
D COPY test TO 'entries.json' (DELIMITER ',');
D select * from 'entries.json';
┌─────────────────────────────┐
│              a              │
│ struct(a varchar, d double) │
├─────────────────────────────┤
│ {'a': bc, 'd': 3.0}         │
└─────────────────────────────┘
```
I intent to fix this in our next release. Casting our nested types to `JSON` should produce valid `JSON` rather than calling our internal `ToString()` method, which is not valid `JSON`.

**Comment by trenta3** [comment]
@lnkuiper Thank you for the detailed explanation. Is there a way I can currently convert the inserted values to proper JSON inside of duckdb, or do I have to write a custom parser (not that big a deal, just asking)?
In my case producing the values takes quite a bit of time and computation (I'm recording certain measures from experiments run) and I'd like to convert those that I already have inserted as opposed to rerun the experiments.

**Comment by lnkuiper** [comment]
If the `STRUCT`s inside of the `JSON` column have a fixed schema, you can do this to convert them to valid `JSON`:
```sql
D CREATE TABLE test (a JSON);
D INSERT INTO test (a) VALUES ({'a': 'bc', 'd': 3.0 });
D select to_json(a::STRUCT(a varchar, d double)) j from test;
┌────────────────────┐
│         j          │
│        json        │
├────────────────────┤
│ {"a":"bc","d":3.0} │ -- Valid JSON!
└────────────────────┘
```
By casting to `STRUCT` first, and then doing `to_json`.

**Comment by lnkuiper** [comment]
Closed via #7366
